charms/istio-pilot/src/charm.py

```diff
@@ -66,6 +65,11 @@
 VIRTUAL_SERVICE_TEMPLATE_FILES = ["src/manifests/virtual_service.yaml.j2"]
 ISTIOCTL_PATH = "./istioctl"
 ISTIOCTL_DEPOYMENT_PROFILE = "minimal"
+INSTALL_FAILED_MSG = (
+    "Failed to install Istio Control Plane."
```

Comment:
Missing whitespace (and I didn't even put two spaces here because I know you prefer one.  Even though two is definitely better :) )
```suggestion
    "Failed to install Istio Control Plane. "
```